core/modules/sql/SqlConnection.cc

```diff
@@ -126,14 +127,29 @@ SqlConnection::~SqlConnection() {
 
 bool
 SqlConnection::connectToDb(SqlErrorObject& errObj) {
-    assert(_connection.get());
-    if (_connection->connected()) return true;
+    assert(_connection != nullptr);
+    static int lastChecked = 0;
```

Comment:
I thought I commented on this. Anyway, all the instances of lastChecked can go, as I don't think there's any point in rate limiting connection attempts. And, as Andy says, lastChecked should not be static but a member variable if you do want to rate limit something. Sharing the variable across all class instances is probably not what you want to do.